harness/determined/cli/workspace.py

```diff
@@ -194,12 +195,12 @@ def create_workspace(args: argparse.Namespace) -> None:
     agent_user_group = _parse_agent_user_group_args(args)
     checkpoint_storage = _parse_checkpoint_storage_args(args)
 
-    if args.cluster_name and not args.namespace:
+    if args.cluster_name and (not args.namespace and not args.auto_create_namespace):
```

Comment:
is there a cleaner way to represent this logic instead of `A & (!B & !C)`? maybe just remove the parentheses?